go/mysql/server_test.go

```diff
@@ -478,11 +478,6 @@ func TestClientFoundRows(t *testing.T) {
 func TestConnCounts(t *testing.T) {
 	th := &testHandler{}
 
-	initialNumUsers := len(connCountPerUser.Counts())
-
-	// FIXME: we should be able to ResetAll counters instead of computing a delta, but it doesn't work for some reason
-	// connCountPerUser.ResetAll()
-
```

Comment:
I've removed this part since I don't think it checks something actually useful and it has issues running the test multiple times. 

We already validate the connection count for the specific user anyway, so we already implicitly check if the user exists in the map. It's not really relevant if the user was just added or already existed or not. It removes a class of test failures that doesn't really imply an actual implementation issue.